### PR TITLE
Arbitrary index options

### PR DESCRIPTION
**Pull request by paterczm**
https://github.com/lightblue-platform/lightblue-core/issues/695

### CODE REVIEW

File: metadata/src/main/java/com/redhat/lightblue/metadata/Index.java [hunk 1]
```diff
@@ -135,4 +137,13 @@ public boolean isCaseInsensitiveKey(Path path) {
                 .filter(IndexSortKey::isCaseInsensitive)
                 .anyMatch(i -> i.getField().equals(path));
     }
+
+    /**
+     *
+     * @return arbitrary index options, can be backend specific
+     */
+    public Properties getOptions() {
```
Comment (bserdar) [inline metadata/src/main/java/com/redhat/lightblue/metadata/Index.java]: You don't really need this. Index extends MetadataObject, and it contains a Properties property. Any unrecognized property will be placed into that properties object as an object map.

File: metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java [hunk 2]
```diff
@@ -328,6 +328,16 @@ public Index parseIndex(T object) {
                 }
                 parseProperties(index, object, INDEX_ELEMENTS);
 
+                // parse arbitrary index options, can be backend specific
+                T options = getMapProperty(object, "options");
```
Comment (bserdar) [inline metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java]: Similarly, you shouldn't need this., Any unrecognized property will be in 'properties'

**Comment by paterczm** [comment]
What is the difference between https://github.com/lightblue-platform/lightblue-mongo/issues/311 and https://github.com/lightblue-platform/lightblue-core/issues/695, if any? I guess I can just close the issue in core.

**Comment by bserdar** [comment]
I believe they're the same

**Pull request closed without merging**